jarbas/dashboard/admin.py

```diff
@@ -305,6 +306,19 @@ def rosies_tweet(self, obj):
     rosies_tweet.short_description = ''
     rosies_tweet.allow_tags = True
 
+    def receipt_link(self, obj):
+        try:
+            receipt_url = obj.get_receipt_url()
+            if receipt_url:
+                return '<a target="_blank" href="{}">📃</a>'.format(receipt_url)
+            else:
+                return ''
+        except Exception:
```

Comment:
What possible exception are we treating here? Generally catching `Exception` is not a best practice IMHO.

But the main point is we can assume [`obj.get_receipt_url()` will return either `None` or a string](https://github.com/datasciencebr/jarbas/blob/master/jarbas/core/models.py#L92-L108), specially because [we have already fetched all possible receipt URLs](https://github.com/datasciencebr/jarbas/blob/master/jarbas/core/management/commands/receipts.py).

Yet if we haven't we don't wanna serialize dozens of requests to the Chamber of Deputies servers per user accessing the dashboard. By our side this will response time probably causing a timeout. By the Chamber of Deputies side they might briefly stop responding to requests from our server. So basically the `.get_receipt_url()` is better used asynchronously, outside the HTTP request/response cycle.

So we could simplify your `receipt_link` method as:

```python
def receipt_link(self, obj):
    if not obj.receipt_url:
        return ''
    return '<a target="_blank" href="{}">📃</a>'.format(obj.receipt_url)
```

What do you think about it?

Reply:
Thats nice!! I associated force=false with not overloading (only problem that I thought that I could cause) and that is true, so... why use the method? hahahah my bad
 I reviewed and understood my mistake.
Thank you, @cuducos

Reply:
C'mon… it's not a mistake ; ) AFAIK you're new to Jarbas, so it's just a matter of context.

Also you pointed a possible ambiguity in the `force=False` argument: it's only used **if** there is already a `receipt_url` fetched. Otherwise it will start a HTTP request to try to fetch it.